tests/daemon/test-machine-id-provider.c

```diff
@@ -102,8 +106,55 @@ static void
 test_machine_id_provider_can_get_id (gboolean     *unused,
                                      gconstpointer dontuseme)
 {
+  const gchar *paths[] = {
+    TESTING_FILE_PATH,
+    NULL
+  };
   EmerMachineIdProvider *id_provider =
-    emer_machine_id_provider_new_full (TESTING_FILE_PATH);
+    emer_machine_id_provider_new_full (paths);
+  uuid_t id;
+  g_assert (emer_machine_id_provider_get_id (id_provider, id));
+  gchar unparsed_id[HYPHENS_IN_ID + FILE_LENGTH];
+  uuid_unparse_lower (id, unparsed_id);
+  gchar* unhypenated_id = unhyphenate_uuid (unparsed_id);
+  g_assert_cmpstr (TESTING_ID, ==, unhypenated_id);
+  g_free (unhypenated_id);
+  g_object_unref (id_provider);
```

Comment:
If you were feeling keen you could add `G_DEFINE_AUTOPTR_CLEANUP_FUNC (EmerMachineIdProvider, g_object_unref);` to `emer-machine-id-provider.h` and then use `g_autoptr(EmerMachineIdProvide) id_provider` above.